src/ApiGateways/Web.Bff.Shopping/aggregator/Infrastructure/GrpcExceptionInterceptor.cs

```diff
@@ -1,41 +1,35 @@
-﻿using Grpc.Core;
-using Grpc.Core.Interceptors;
-using Microsoft.Extensions.Logging;
-using System.Threading.Tasks;
+﻿namespace Microsoft.eShopOnContainers.Web.Shopping.HttpAggregator.Infrastructure;
 
-namespace Microsoft.eShopOnContainers.Web.Shopping.HttpAggregator.Infrastructure
+public class GrpcExceptionInterceptor : Interceptor
 {
-    public class GrpcExceptionInterceptor : Interceptor
+    private readonly ILogger<GrpcExceptionInterceptor> _logger;
+
+    public GrpcExceptionInterceptor(ILogger<GrpcExceptionInterceptor> logger)
     {
-        private readonly ILogger<GrpcExceptionInterceptor> _logger;
+        _logger = logger;
+    }
 
-        public GrpcExceptionInterceptor(ILogger<GrpcExceptionInterceptor> logger)
-        {
-            _logger = logger;
-        }
+    public override AsyncUnaryCall<TResponse> AsyncUnaryCall<TRequest, TResponse>(
+        TRequest request,
+        ClientInterceptorContext<TRequest, TResponse> context,
+        AsyncUnaryCallContinuation<TRequest, TResponse> continuation)
+    {
+        var call = continuation(request, context);
 
-        public override AsyncUnaryCall<TResponse> AsyncUnaryCall<TRequest, TResponse>(
-            TRequest request,
-            ClientInterceptorContext<TRequest, TResponse> context,
-            AsyncUnaryCallContinuation<TRequest, TResponse> continuation)
-        {
-            var call = continuation(request, context);
+        return new AsyncUnaryCall<TResponse>(HandleResponse(call.ResponseAsync), call.ResponseHeadersAsync, call.GetStatus, call.GetTrailers, call.Dispose);
+    }
 
-            return new AsyncUnaryCall<TResponse>(HandleResponse(call.ResponseAsync), call.ResponseHeadersAsync, call.GetStatus, call.GetTrailers, call.Dispose);
+    private async Task<TResponse> HandleResponse<TResponse>(Task<TResponse> t)
+    {
+        try
+        {
+            var response = await t;
```

Comment:
```suggestion
    private async Task<TResponse> HandleResponse<TResponse>(Task<TResponse> task)
    {
        try
        {
            var response = await task;
```

Nit: let's spell this out.

Reply:
ok